pyfar/classes/audio.py

```diff
@@ -1021,7 +1039,122 @@ def power(data: tuple, domain='freq'):
+    :py:func:`Signal`, :py:func:`TimeData`, and :py:func:`FrequencyData`.
+
+    Parameters
+    ----------
+    data : tuple of the form (data_1, data_2, ..., data_N)
+        Data to be multiplied. Can contain pyfar audio objects or array likes.
+        The data is `broadcasted
+        <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+        to the data with the highest number of channel dimensions. Pyfar audio
+        objects can not be mixed, e.g., :py:func:`TimeData` and
+        :py:func:`FrequencyData` objects do not work together. See
+        :py:mod:`arithmetic operations <pyfar._concepts.arithmetic_operations>`
+        for possible combinations of Signal FFT normalizations.
+    domain : ``'time'``, ``'freq'``, optional
+        Flag to indicate if the operation should be performed in the time or
+        frequency domain. If working in the frequency domain, the FFT
+        normalization is removed before the operation (See
+        :py:func:`~pyfar.dsp.fft.normalization`). The default is ``'freq'``.
+    axes : list of 3 tuples
+        A list of tuples with the indices of the channels the
+        multiplication should operate on. The signature is (i,j),(j,k)->(i,k),
+        so the base elements of the multiplication are two-dimensional
+        matrices along the time/frequency dimension taken to be stored in the
+        indicated channels of the result. Negative values in ``axes`` refer to
+        ``cshape``, as the frequency/time axis is excluded. For instance, to
+        multiply matrices residing (stacked) in the last two channel
+        dimensions to be stored in the last two channel dimensions, this is
+        indicated by ``[(-2, -1), (-2, -1), (-2, -1)]``, which is the default.
+        See notes and examples below.
+
+    Returns
+    -------
+    results : Signal, TimeData, FrequencyData, numpy array
+        Result of the operation as numpy array, if `data` contains only array
+        likes and numbers. Result as pyfar audio object if `data` contains an
+        audio object. The `fft_norm` is ``'none'`` if all FFT norms are
+        ``'none'``. Otherwise the first `fft_norm` that is not ``'none'`` is
+        used.
+
+    Notes
+    -----
+    The behavior depends on the data in the following way.
+
+    - In case of more than two arguments in ``data``, ``axes`` refers to all
+      partial operations.
+    - If the ``cshape`` of the arguments in data is 2-D, the arguments are
+      multiplied like conventional matrices along the time/frequency dimension.
+    - If either ``cshape`` is N-D, N > 2, the channels are treated as a stack
+      of matrices residing in the last two indexes and broadcasted accordingly.
+    - If the first ``cshape`` is 1-D, it is promoted to a 2-D ``cshape`` by
+      prepending a 1 to its dimensions.
+    - If the second ``cshape`` is 1-D, it is promoted to a 2-D ``cshape`` by
+      appending a 1 to its dimensions.
```

Comment:
What is the idea behind this case dependent handling? I'm wondering if this would mess with (AB)C = A(BC). Should we maybe insist that shapes match right away and not do any prepending/appending?

Reply:
Basically, it refers to the fact that - depending on the position of the signal in operation - a 1-D signal is either extended in the "row" or "column" channel dimension for the matrix multiplication before broadcasting. I'm not sure if this answers your question?

I would prefer allowing signals with a 1-D cshape. But we can also refer the ``reshape``, if we want to restrict.

Reply:
I think the numpy default handling is very reasonable in this case.